src/Microsoft.ML.Data/Commands/CrossValidationCommand.cs

```diff
@@ -328,11 +328,14 @@ private string GetSplitColumn(IChannel ch, IDataView input, ref IDataView output
                 {
                     ch.Info("Hashing the stratification column");
                     var origStratCol = stratificationColumn;
-                    int tmp;
-                    int inc = 0;
-                    while (input.Schema.TryGetColumnIndex(stratificationColumn, out tmp))
-                        stratificationColumn = string.Format("{0}_{1:000}", origStratCol, ++inc);
-                    output = new HashingEstimator(Host, origStratCol, stratificationColumn, 30).Fit(input).Transform(input);
+                    stratificationColumn = input.Schema.GetTempColumnName("strat");
+
+                    // HashingEstimator currently handles all primitive types except for DateTime, DateTimeOffset and TimeSpan.
+                    var itemType = type.GetItemType();
+                    if (itemType is DateTimeDataViewType || itemType is DateTimeOffsetDataViewType || itemType is TimeSpanDataViewType)
+                        input = new TypeConvertingTransformer(Host, origStratCol, DataKind.Int64, origStratCol).Transform(input);
+
+                    output = new HashingEstimator(Host, stratificationColumn, origStratCol, 30).Fit(input).Transform(input);
                 }
```

Comment:
I realize the previous code too used 30 bits. But the default seems to be 31 bits. Any reason why we have 30 here? If there is a specific reason for 30 bits, can you please add a note to that in the code?

Reply:
I am actually not sure why 30 was chosen. If you think it's important I can try changing it back to 31.

---
In reply to: [386729669](https://github.com/dotnet/machinelearning/pull/4828#discussion_r386729669) [](ancestors = 386729669)

Reply:
I have no reason to say it should change to 31 either :-). But since it is is used in a couple of places, can you please make it a named constant?

---
In reply to: [425102125](https://github.com/dotnet/machinelearning/pull/4828#discussion_r425102125) [](ancestors = 425102125,386729669)